### PR TITLE
Fix node.is_seed logic

### PR DESCRIPTION
**Pull request by fruch**
* Remove as much of the defaulting to seed as node[0]
* move to using node.ip_address(), which is automaticlly selecting public/private ips (based on config or if it's multi-region setup)

### CODE REVIEW

**Comment by fruch** [comment]
@roydahan do we want to port this fix to other branches too ?

**Comment by roydahan** [comment]
2019.1 is enough.
However, I would like to test it with master first, so please do not backport till you get an answer from me.
@bentsi reminder for you to check with me on Sunday.

File: sdcm/cluster.py [hunk 1]
```diff
@@ -1846,14 +1846,13 @@ def disable_client_encrypt(self):
             node.disable_client_encrypt()
 
     def get_seed_nodes(self):
-        seed_nodes_private_ips = self.get_seed_nodes_private_ips()
+        seed_nodes_private_ips = self.get_seed_nodes_from_node()
```
Comment (bentsi) [inline sdcm/cluster.py]: `seed_nodes_private_ips` -> `seed_nodes_ips`

File: sdcm/cluster.py [hunk 2]
```diff
@@ -1815,7 +1815,7 @@ def get_scylla_args(self):
         return self.params.get('append_scylla_args_oracle') if self.name.find('oracle') > 0 else\
             self.params.get('append_scylla_args')
 
-    def get_seed_nodes_private_ips(self):
+    def get_seed_nodes_from_node(self):
         if self.seed_nodes_private_ips is None:
```
Comment (bentsi) [inline sdcm/cluster.py]: seed_nodes_private_ips -> seed_nodes_ips

**Pull request merged**